src/examples/java/software/amazon/encryption/s3/examples/MultipartUploadExample.java

```diff
@@ -0,0 +1,192 @@
+                    .bucket(BUCKET)
+                    .key(objectKey)
+                    .uploadId(initiateResult.uploadId())
+                    .partNumber(partsSent)
+                    .build();
+
+            final InputStream partInputStream = new ByteArrayInputStream(outputStream.toByteArray());
+
+            // Upload all the different parts of the object
+            UploadPartResponse uploadPartResult = v3Client.uploadPart(uploadPartRequest,
+                    RequestBody.fromInputStream(partInputStream, partInputStream.available()));
+
+            // We need to add eTag's of all CompletedParts before calling CompleteMultipartUpload.
+            partETags.add(CompletedPart.builder()
+                    .partNumber(partsSent)
+                    .eTag(uploadPartResult.eTag())
+                    .build());
+
+            outputStream.reset();
+            bytesSent = 0;
+            partsSent++;
+        }
+        inputStream.close();
+
+        // Set sdkPartType to SdkPartType.LAST for last part of the multipart upload.
+        //
+        // Note: Set sdkPartType parameter to SdkPartType.LAST for last part is required for Multipart Upload in S3EncryptionClient to call `cipher.doFinal()`
+        UploadPartRequest uploadPartRequest = UploadPartRequest.builder()
+                .bucket(BUCKET)
+                .key(objectKey)
+                .uploadId(initiateResult.uploadId())
+                .partNumber(partsSent)
+                .sdkPartType(SdkPartType.LAST)
+                .build();
+
+        // Upload the last part multipart upload to invoke `cipher.doFinal()`
+        final InputStream partInputStream = new ByteArrayInputStream(outputStream.toByteArray());
+        UploadPartResponse uploadPartResult = v3Client.uploadPart(uploadPartRequest,
+                RequestBody.fromInputStream(partInputStream, partInputStream.available()));
+
+        partETags.add(CompletedPart.builder()
+                .partNumber(partsSent)
+                .eTag(uploadPartResult.eTag())
+                .build());
+
+        // Finally call completeMultipartUpload operation to tell S3 to merge all uploaded
+        // parts and finish the multipart operation.
+        v3Client.completeMultipartUpload(builder -> builder
+                .bucket(BUCKET)
+                .key(objectKey)
+                .uploadId(initiateResult.uploadId())
+                .multipartUpload(partBuilder -> partBuilder.parts(partETags)));
+
+        // Call getObject to retrieve and decrypt the object from S3
+        ResponseBytes<GetObjectResponse> result = v3Client.getObjectAsBytes(builder -> builder
+                .bucket(BUCKET)
+                .key(objectKey));
+
+        // Asserts
+        String inputAsString = IoUtils.toUtf8String(new BoundedInputStream(fileSizeLimit));
+        String outputAsString = IoUtils.toUtf8String(result.asInputStream());
+        assertEquals(inputAsString, outputAsString);
```

Comment:
we should be doing something like: 

```
        assertTrue(IOUtils.contentEquals(objectStreamForResult, output));
```

to avoid buffering the entire object into `String`

Reply:
ack!